### PR TITLE
Records: Clerical expansion

### PR DESCRIPTION
**Pull request by jlsnow301**
## About The Pull Request
A few changes to sec records geared towards making it obvious how and why someone is wanted.

- You can no longer set someone to arrest via console unless they have a valid prior. 
- Setting players to arrest via HUD creates a crime entry for that person.
- Priors can be edited by the author and qualified personnel (armory access).
- Priors can be invalidated by qualified personnel. This redacts the info from huds and rapsheets.
- Invalidating the last crime sets a player to discharged
- Much like real life, you can no longer outright delete crimes. This prevents gaming the arrest system by adding/deleting.
- Deleting an individual record in a sec console removes it from the manifest (@JohnFulpWillard) this was previous behavior
- Purging records in sec requires higher access

## Why It's Good For The Game
Have you ever brought a wanted person into the brig only to find there's no reasons listed and no one in sec will explain why they're wanted?
## Changelog
:cl:
fix: Deleting a sec record removes it from the manifest again
qol: It's now easier to see security notes via HUD.
qol: It's now easier to see why someone is set to arrest.
qol: Crime authors and armory access can edit crimes.
balance: Setting players to arrest requires a valid crime.
balance: Armory access can invalidate priors.
balance: Setting players to arrest via HUD creates a crime for the suspect.
/:cl:

### CODE REVIEW

**Comment by VladinXXV** [comment]
How does this work with ninja setting everyone to arrest?

I'm not entirely sure about this, but I'm choosing to remain hopeful it might actually get people to fill out records properly.

**Comment by jlsnow301** [comment]
> How does this work with ninja setting everyone to arrest?

As of yet he just sets it regardless. I can maybe make it create a fake crime, but how easily that would be meta'd

**Comment by VladinXXV** [comment]
> > How does this work with ninja setting everyone to arrest?
> 
> As of yet he just sets it regardless. I can maybe make it create a fake crime, but how easily that would be meta'd

I mean, meta'd how? AI already gets notified when you start the process, and when everyone gets set to arrest it's already pretty known to assume ninja. That said, maybe it's still better without a fake crime, just for the sake of not mucking up records and leading to unfair punishments to players who appear to be on a crime spree.

**Comment by jlsnow301** [comment]
> I mean, meta'd how? 

I can't make it any less obvious, and making a fake crime with random details would probably be memorized by security players. 

> maybe it's still better without a fake crime, just for the sake of not mucking up records

Agree for now, we'll see if/how it pans out

**Comment by Pickle-Coding** [comment]
Maybe fake crimes could work like Poly, where it picks randomly from real set crimes people used.

**Comment by jlsnow301** [comment]
I want to use the word clerical ok

File: code/game/machinery/computer/medical.dm [hunk 1]
```diff
@@ -40,7 +40,7 @@
 	var/list/data = list()
 
 	var/has_access = authenticated && isliving(user)
-	data["authenticated"] = authenticated
+	data["authenticated"] = authenticated || isAdminGhostAI(user)
```
Comment (JohnFulpWillard) [inline code/game/machinery/computer/medical.dm:43]: Would this work since ``has_access`` would still be false?

Comment (jlsnow301) [reply code/game/machinery/computer/medical.dm:43]: good catch

File: code/datums/records/record.dm [hunk 2]
```diff
@@ -218,10 +218,16 @@
...
+		if(crime.valid)
+			final_paper_text += "<tr><td>[crime.name]</td>"
+			final_paper_text += "<td>[crime.details]</td>"
+			final_paper_text += "<td>[crime.author]</td>"
+			final_paper_text += "<td>[crime.time]</td>"
+		else
+			final_paper_text += "<td>--REDACTED--</td>"
+			final_paper_text += "<td>--REDACTED--</td>"
+			final_paper_text += "<td>--REDACTED--</td>"
+			final_paper_text += "<td>--REDACTED--</td>"
```
Comment (san7890) [inline code/datums/records/record.dm:230]:
```suggestion
			for(var/i in 1 to 4)
				final_paper_text += "<td>--REDACTED--</td>"
```

:devilcat:

Comment (jlsnow301) [reply code/datums/records/record.dm:230]: maximum efficiency

**Comment by LSommers** [comment]
It would be nice if Warden/HoS could delete or at least modify entries. As warden, I often have to delete duplicate entries, delete and remake incorrectly made ones, misspelled or blank crime details, people using the crime field for details, etc.

**Comment by jlsnow301** [comment]
> It would be nice if Warden/HoS could delete or at least modify entries.

They can

**Comment by LSommers** [comment]
No idea if there's a newer PR I should make note of this on, but fines cannot be nullified from a sec console and notes are nonfunctional (doesn't save when typed up).

**Comment by optimumtact** [comment]
@jlsnow301 epic fail

**Comment by jlsnow301** [comment]
> notes are nonfunctional (doesn't save when typed up).

did u just report a stale bug 🗞️ #73363

Screenshot 2023-03-05 231343

the other part i'll look into

**Comment by LSommers** [comment]
> did u just report a stale bug 🗞️ #73363

Hell if I know. I figured a whole actual new issue was probably overkill.
Last time I tried to set notes they didn't persist, I'll check again though.

**Pull request merged**